ouster-lidar/ouster-ros
# SW-5657: Back port SW-5466 to ROS-1

## Related Issues & PRs
- parent #216 
- resolves #15 
- resolves #29
- resolves #210 
- resolves #97
- related #217
- related #223
- resolves #231 (adds a note to CHANGLOG.rst indicating that the default `ptp_utc_tai_offset` forms a breaking change).  

## Summary of Changes
- added the ability to customize the published point clouds(s) to velodyne point cloud format and
  other common pcl point types.
- made ouster_image_nodelet operate separately from ouster_cloud_nodelet
- added a note to `CHANGLOG.rst` indicating that the default `ptp_utc_tai_offset` forms a breaking change.
- require C++17 for the ROS1 driver starting with v0.11 (C++14 :no_good:)

## Validation (TBD)
- Run the driver with default parameters and verify no observable behavioral changes.
- Change `point_type` parameter to `native` and verify that it reflects the same point cloud format that we get from the sensor.
- Change `point_type` parameter to `xyz` and verify that the published point cloud format only has `xyz` fields, regardless of the underlying sensor format.
- Change `point_type` parameter to `xyzi` and verify that the published point cloud format only has `xyz` fields plus intensity, regardless of the underlying sensor format (you should get a warning when the low data mode is selected for this mode).
- Change `point_type` parameter to `xyzir` and verify that the published point cloud format only has `xyz` fields plus intensity plus ring, regardless of the underlying sensor format (you should get a warning when the low data mode is selected for this mode).
